contents/tutorials/nextjs-bootstrap-flags.md

```diff
@@ -0,0 +1,290 @@
+---
+title: How to use Next.js middleware to bootstrap feature flags
+date: 2023-07-03
+author: ["ian-vanagas"]
+showTitle: true
+sidebar: Docs
+featuredImage: ../images/tutorials/banners/nextjs-analytics.png
+tags: ["configuration", "feature flags"]
+---
+
+Next.js has middleware enabling you to run functions between requests and responses. We can use this to bootstrap feature flags on page load and make them available immediately. This is useful for redirecting, showing components without flickering, avoiding layout changes, and more.
```

Comment:
Good intro. Short and clear

Im not sure if "without having to make an additional requests" is technically correct, or if theres a better way to phrase it

```suggestion
[Next.js middleware](https://nextjs.org/docs/app/building-your-application/routing/middleware) enables you to run functions between requests and responses. We can use this to [bootstrap feature flags](/docs/feature-flags/bootstrapping-and-local-evaluation) on page load and make them available immediately – without having to make an additional requests. This is useful for redirecting, showing components without flickering, avoiding layout changes, and more.
```